src/style.rs

```diff
@@ -340,17 +356,41 @@ impl ProgressStyle {
                                 align: *align,
                                 truncate: *truncate,
                             };
-                            match style {
-                                Some(s) => cur
-                                    .write_fmt(format_args!("{}", s.apply_to(padded)))
-                                    .unwrap(),
-                                None => cur.write_fmt(format_args!("{padded}")).unwrap(),
+                            if self.json_strings {
+                                match style {
+                                    Some(s) => cur.push_str(&json::stringify(format!(
+                                        "{}",
+                                        s.apply_to(padded)
+                                    ))),
+                                    None => cur.push_str(&json::stringify(format!("{padded}"))),
+                                }
+                            } else {
+                                match style {
+                                    Some(s) => cur
+                                        .write_fmt(format_args!("{}", s.apply_to(padded)))
+                                        .unwrap(),
+                                    None => cur.write_fmt(format_args!("{padded}")).unwrap(),
+                                }
+                            }
+                        }
+                        None => {
+                            if self.json_strings {
+                                match style {
+                                    Some(s) => cur.push_str(&json::stringify(format!(
+                                        "{}",
+                                        s.apply_to(&buf)
+                                    ))),
+                                    None => cur.push_str(&json::stringify(buf.clone())),
+                                }
+                            } else {
+                                match style {
+                                    Some(s) => {
+                                        cur.write_fmt(format_args!("{}", s.apply_to(&buf))).unwrap()
+                                    }
+                                    None => cur.push_str(&buf),
+                                }
```

Comment:
(I'm personally not 100% happy with the complexity here, so if someone has better ideas...)